### ISSUE TITLE
single mode

### ISSUE THREAD

**Issue by blue-butterfly** [tensorlayer/hyperpose#177]
Hi, I want to train the program with the 'single' gpu mode. but when I train, the command 'nvidia-smi' presents not any gpu info. I don't know why,  could you help me?

Total Loss at iteration 2 / 600000 is: 2936.142578125 Learning rate 4.000000e-05 l2_loss 3.060536e+00 Took: 16.417922019958496s
Network# 0 For Branch 1 Loss: 921.5547
Network# 1 For Branch 2 Loss: 64.268234
Network# 2 For Branch 1 Loss: 907.3777
Network# 3 For Branch 2 Loss: 64.260765
Network# 4 For Branch 1 Loss: 918.7116
Network# 5 For Branch 2 Loss: 64.337036
Network# 6 For Branch 1 Loss: 930.89404
Network# 7 For Branch 2 Loss: 64.31972
Network# 8 For Branch 1 Loss: 908.6487
Network# 9 For Branch 2 Loss: 64.28439
Network# 10 For Branch 1 Loss: 893.205
Network# 11 For Branch 2 Loss: 64.3025
Total Loss at iteration 3 / 600000 is: 2272.997314453125 Learning rate 4.000000e-05 l2_loss 3.060541e+00 Took: 16.712655067443848s
Network# 0 For Branch 1 Loss: 671.1072
Network# 1 For Branch 2 Loss: 104.39488
Network# 2 For Branch 1 Loss: 638.52936
Network# 3 For Branch 2 Loss: 104.39094
Network# 4 For Branch 1 Loss: 671.3517
Network# 5 For Branch 2 Loss: 104.81377
Network# 6 For Branch 1 Loss: 706.5357
Network# 7 For Branch 2 Loss: 104.61961
Network# 8 For Branch 1 Loss: 623.45953
Network# 9 For Branch 2 Loss: 104.59555
Network# 10 For Branch 1 Loss: 601.4993
Network# 11 For Branch 2 Loss: 104.57595
Total Loss at iteration 4 / 600000 is: 1452.595703125 Learning rate 4.000000e-05 l2_loss 3.060560e+00 Took: 15.728559494018555s

Processes:                                                       GPU Memory |
|  GPU       PID   Type   Process name                             Usage      |
|=============================================================================|
|  No running processes found

**Comment by abaybektursun** [comment]
The single mode actually trains on CPU from what I gather

**Comment by elmuz** [comment]
> The single mode actually trains on CPU from what I gather

No, it doesn't! Just replace _tensorflow_ with tensorflow-gpu_ in the requirements.txt.

**Comment by abaybektursun** [comment]
@elmuz I don't know then what I was doing wrong. With tf-gpu installed single mode, mobilenet would take 90s every step. I tried to change to parallel (4GPUs) it satrted doing 2-3s per step. That's a ridiculous speedup, so I assumed single was training on CPU.

**Comment by abaybektursun** [comment]
@elmuz 
Check out this post from 2018:
I lifted the train.py
"
# os.environ["CUDA_DEVICE_ORDER"] = "PCI_BUS_ID"
# os.environ["CUDA_VISIBLE_DEVICES"] = "0"
"
Comment

Training in single simgle mode
run
Python3 train.py

Then use nvidia-smi to view the gpu usage, and did not see the training related gpu usage information.

_Originally posted by @drt040312 in https://github.com/tensorlayer/openpose-plus/issues/13#issuecomment-442729229_

**Comment by elmuz** [comment]
It worked to me. If you install tensorflow-gpu and train in single mode you will see through nvidia-smi the use of a single GPU, even if you have more on your machine. In my case with 1 GPU Nvidia 1080ti it goes around 1.5sec/iter.

**Comment by blue-butterfly** [comment]
Yes，I install both tensorflow and tensorflow-gpu，but the progress uses the tensorflow defaultly。When I uninstall tensorflow，the tensorflow-gpu works.

**Comment by abaybektursun** [comment]
@elmuz Can you please check what FPS you are getting when running `profile.sh` with mobilenet??

**Comment by YinXiaojieCSDN** [comment]
> @elmuz This is somewhat unrelated, but can you please check what FPS you are getting when running `profile.sh` with mobilenet??

Hello, I want to know have you solved it?
